R/ROOT/build_tarballs.jl

```diff
@@ -0,0 +1,216 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+
+using BinaryBuilder, Pkg
+
+name = "ROOT"
+version = v"6.32.8"
+
+rootgithash = Dict(v"6.32.6" => "5380676d1f0ed3055048f305de917572f976a090",
+                   v"6.32.8" => "a1748a069d2d5f27ec528cfd89b54b12dcba793e")
+
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/root-project/root.git", rootgithash[version]),
+    DirectorySource("./bundled")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE
+if echo "$target" | grep -q musl; then #build wih musl library
+# Disabling what is not working with musl:
+    CMAKE_EXTRA_OPTS=(-Ddavix=OFF -Dxrootd=OFF -Dssl=OFF -Druntime_cxxmodules=OFF -Droofit=OFF)
+else
+    CMAKE_EXTRA_OPTS=(-Druntime_cxxmodules=OFF)
+fi
+
+#Uncomment for a minimal build for debugging purposes
+#CMAKE_EXTRA_OPTS+=(-Dclad=OFF -Dhtml=OFF -Dwebgui=OFF -Dcxxmodules=OFF -Dproof=OFF -Dtmva=OFF -Drootfit=OFF -Dxproofd=OFF -Dxrootd=OFF -Dssl=OFF -Dpyroot=OFF -Dtesting=OFF -Droot7=OFF -Dspectrum=OFF -Dunfold=OFF -Dasimage=OFF -Dgviz=OFF -Dfitiso=OFF -Dcocoa=OFF -Dopengl=OFF -Dproof=OFF -Dxml=OFF -Dgfal=OFF -Dmpi=OFF)
+
+export SYSTEM_INCLUDE_PATH="`g++ --sysroot="/opt/$target/$target/sys-root" -E -x c++ -v /dev/null  2>&1  | awk '{gsub(\"^ \", \"\")} /End of search list/{a=0} {if(a==1){s=s d $0;d=":"}} /#include <...> search starts here/{a=1} END{print s}'`"
+
+# build-in compilation of the libAfterImage library needs this directory
+mkdir -p /tmp/user/0
+
+cd /
+
+cd "$WORKSPACE"
+
+# For the cross-compiling, LLVM needs to compile the llvm-tblgen tool
+# for the host. The ROOT LLVM code tree does not include the test and
+# benchmark directories cmake will need to configure for the host for
+# this bootstrap. Therefore, we need to disable the build of the test and
+# benchmarks.
+sed -i 's/\(option(LLVM_INCLUDE_TESTS[[:space:]].*[[:space:]]\)on)\(.*\)/\1OFF)\2/i
+        s/\(option(LLVM_INCLUDE_BENCHMARKS[[:space:]].*[[:space:]]\)on)\(.*\)/\1OFF)\2/i' \
+        srcdir/root/interpreter/llvm-project/llvm/CMakeLists.txt
+echo "set(CXX_STANDARD 17)" >> srcdir/root/interpreter/llvm-project/llvm/CMakeLists.txt
+
+# N llvm links. LLVM link command needs 15GB
+njobs=$((2*`nproc`))
+LLVM_PARALLEL_LINK_JOBS=`grep MemTot /proc/meminfo  | awk '{a=int($2/15100000); if(a>'"$njobs"') a='"$njobs"'; if(a<1) a=1; print a;}'`
```

Comment:
I feel like you need to do all of this because you were using setting `njobs` incorrectly.

Reply:
This follows the recommendations from https://llvm.org/docs/CMake.html#frequently-used-llvm-related-variables, section LLVM_PARALLEL_{COMPILE,LINK}_JOBS. Machines have often less than 15GB RAM per core, so it it is also needed without the factor 2.